### ISSUE TITLE
Typescript error

### ISSUE THREAD

**Issue by MoJo2600** [jamesmbourne/aws4-axios#8]
Maybe this is a beginners question, but I'm following the example on how to use the aws4-axios interceptor. But I'm getting a TypeScript error.

```
    const interceptor = aws4Interceptor({
      region: "eu-west-2",
      service: "execute-api"
    });

    axios.interceptors.request.use(interceptor);
```

The error is:
``` 
error TS2345:
Argument of type '(config: AxiosRequestConfig) => AxiosRequestConfig' is not assignable to parameter of type '(value: AxiosRequestConfig) => AxiosRequestConfig | Promise<AxiosRequestConfig>'.
Types of parameters 'config' and 'value' are incompatible.
```

Does anybody know what I'm doing wrong?

**Comment by youngpayters** [comment]
Got the same. Looks like it is not compatible with axios 0.20.0

**Comment by MoJo2600** [comment]
Yes, i reverted back :/

**Comment by youngpayters** [comment]
@jamesmbourne any chance you can take a look here?

**Comment by cam8001** [comment]
I'm seeing the same with Axios `0.21.0` (current version).

As a temp fix you can change the version of axios in your `package.json` to `0.19.0` and things seem to work fine.

**Comment by jamesmbourne** [comment]
Will try and get a new version out this week 👍

**Comment by jamesmbourne** [comment]
Axios bump in #22

**Comment by jamesmbourne** [comment]
This should be addressed in #23

**Comment by jamesmbourne** [comment]
I have published `v2.0.1` which is compatiable with `axios@0.21.0`. Please try upgrading to that and let me know if you have any issues.

```json
"dependencies": {
  "aws4-axios": "^2.0.1",
  "axios": "^0.21.0"
}
```
